cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/templates/etl/batch/ETLMapReduce.java

```diff
@@ -136,15 +136,17 @@ public void initialize(MapReduceContext context) throws Exception {
       List<StageMetrics> stageMetrics = Lists.newArrayListWithCapacity(stageList.size() + 2);
       transforms = Lists.newArrayListWithCapacity(stageList.size());
 
-      BatchSource source = context.newPluginInstance(sourceId);
-      source.initialize(context.getPluginProperties(sourceId));
+      BatchSource source = context.newPluginInstance(sourcePluginId);
+      BatchSourceContext batchSourceContext = new MapReduceSourceContext(context, mapperMetrics, sourcePluginId);
```

Comment:
Yes. Earlier batch sources/sinks were only configuring hadoop job and nothing else. now, since they have initialize, transform, destroy methods, emitting custom metrics would be useful.